src/main/scala/com/gu/fulfilmentLookup/AwsS3Client.scala

```diff
@@ -0,0 +1,37 @@
+package com.gu.fulfilmentLookup
+
+import java.io.InputStream
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.s3.model.GetObjectRequest
+import purecsv.unsafe.CSVReader
+import scala.io.Source
+import scala.util.Try
+
+trait CsvClient {
+  def getDeliveryRows(s3Stream: InputStream): Try[List[DeliveryRow]]
+  def getDeliveryRowsFromS3(bucketName: String, subFolder: String, fileName: String): Try[List[DeliveryRow]]
+}
+
+object AwsS3Client extends CsvClient with Logging {
+
+  val s3Client = AmazonS3ClientBuilder.defaultClient()
+
+  override def getDeliveryRows(s3Stream: InputStream): Try[List[DeliveryRow]] = {
+    val csvAsString = Source.fromInputStream(s3Stream).mkString
+    val parseAttempt = Try(CSVReader[DeliveryRow].readCSVFromString(csvAsString, skipHeader = true))
```

Comment:
actually the mkString pulls everything into one long string in memory, so you neednt keep the stream open when you're parsing it.
In fact, the way you've split things up may make more sense if you had

1. turn the s3 file name into a String of the content
1. turn the content into a list of DeliveryRows

that way, the second step doesn't have any external dependencies.
What I'm really suggesting is move the code that deals with the stream into getDeliveryRowsFromS3 and just keep the parseattempt line in getDeliveryRows